docs/dump_javadump.md

```diff
@@ -27,8 +27,519 @@
+NULL
+1STSEGTOTAL    Total memory:                   268435456 (0x0000000010000000)
+1STSEGINUSE    Total memory in use:                34704 (0x0000000000008790)
+1STSEGFREE     Total memory free:              268400752 (0x000000000FFF7870)
+1STSEGLIMIT    Allocation limit:               268435456 (0x0000000010000000)
+NULL           
+1STSEGTYPE     JIT Data Cache
+NULL           segment            start              alloc              end                type       size
+1STSEGMENT     0x00007FF4F0096668 0x00007FF4CC553030 0x00007FF4CC753030 0x00007FF4CC753030 0x00000048 0x0000000000200000
+NULL
+1STSEGTOTAL    Total memory:                     2097152 (0x0000000000200000)
+1STSEGINUSE    Total memory in use:              2097152 (0x0000000000200000)
+1STSEGFREE     Total memory free:                      0 (0x0000000000000000)
+1STSEGLIMIT    Allocation limit:               402653184 (0x0000000018000000)
+NULL           
+1STGCHTYPE     GC History  
+NULL           
+```
+In the example, the GC History (`1STGCHTYPE`) section is blank. This section is populated if a garbage collection cycle occurred in
+a VM that is being diagnosed with the trace facility.
+
+## LOCKS
+
+This section of the Java dump provides information about locks, which protect shared resources from being accessed by more than one entity at a time. The information is essential in a deadlock situation, where two threads attempt to synchronize on an object and lock an instance of a class. Precise information is recorded about the threads that are causing the problem, which enables you to identify the root cause.
+
+The following example shows a typical LOCKS section, where no deadlocks existed at the time the dump was triggered. For clarity, the following example shows a shortened version of this section, where `...` indicates that lines are removed:
+
+```
+NULL           ------------------------------------------------------------------------
+0SECTION       LOCKS subcomponent dump routine
+NULL           ===============================
+NULL           
+1LKPOOLINFO    Monitor pool info:
+2LKPOOLTOTAL     Current total number of monitors: 3
+NULL           
+1LKMONPOOLDUMP Monitor Pool Dump (flat & inflated object-monitors):
+2LKMONINUSE      sys_mon_t:0x00007FF4B0001D78 infl_mon_t: 0x00007FF4B0001DF8:
+3LKMONOBJECT       java/lang/ref/ReferenceQueue@0x00000000FFE26A10: <unowned>
+3LKNOTIFYQ            Waiting to be notified:
+3LKWAITNOTIFY            "Common-Cleaner" (J9VMThread:0x0000000000FD0100)
+NULL           
+1LKREGMONDUMP  JVM System Monitor Dump (registered monitors):
+2LKREGMON          Thread global lock (0x00007FF4F0004FE8): <unowned>
+2LKREGMON          &(PPG_mem_mem32_subAllocHeapMem32.monitor) lock (0x00007FF4F0005098): <unowned>
+2LKREGMON          NLS hash table lock (0x00007FF4F0005148): <unowned>
+...
+NULL           
+```
+
+## THREADS
```

Comment:
There is no explanation of the THREADS section.

Reply:
Yep, WIP. Will get back to it soon. Thanks for the SHARED CLASSES output.

Reply:
SHARED CLASSES and THREADS sections now in place. I think that the topic now contains a relatively complete set of information. Do we know if the cgroup PR will go in to 0.10.0 please? If it will not, I'll comment out the parts that relate to it and I think this is good to merge.

What I'd like to do as a separate task is take 2 or 3 specific scenarios (hopefully from service) and create troubleshooting examples. E.g. I got a gpf, what can javadump tell me about the root cause? If we can supplement this topic with those examples, I think the info here is as good as it was in the J9 docs. If you have any thoughts on what the scenarios should be, please let me know.

Reply:
cgroup PR is still under review and blocked on the docs being ready. It doesn't need to go into 0.10.0, but it depends on when the review comments are suitably addressed and the branch for 0.10.0 occurs. In theory it should go in after 0.10.0 but no guarantees.

Reply:
gpf or assert, user signal, OOM, deadlock maybe.

Reply:
Thanks for the suggestions on scenarios.

For cgroup, I think the docs are ready .... I've got it in the release notes and what's new, plus this javadump topic contains a mention and example output. One of the developers is writing a blog about it, which I might be able to extract more information from later on. If I can get this topic merged I can close the doc issue. Let me know ....

Reply:
The cgroup review still needs to close before it can be merged, and the review comments haven't yet been addressed. We were hoping to branch for 0.10.0 tomorrow, but seems likely we won't be ready.

Reply:
cgroup information removed for 0.10.0